p2p/net/swarm/swarm_dial.go

```diff
@@ -371,87 +373,84 @@ func (s *Swarm) dialAddrs(ctx context.Context, d *conn.Dialer, p peer.ID, remote
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel() // cancel work when we exit func
 
-	foundConn := make(chan struct{})
-	conns := make(chan conn.Conn, len(remoteAddrs))
+	conns := make(chan conn.Conn)
```

Comment:
if this gets buffered, then open connections just sit in the channel and never get closed